### PR TITLE
add metagem bioconda recipe

### PR DESCRIPTION
**Pull request by franciscozorrilla**
Description: An easy-to-use workflow for generating context specific genome-scale metabolic models and predicting metabolic interactions within microbial communities directly from metagenomic data

GitHub: https://github.com/franciscozorrilla/metaGEM

Paper: https://academic.oup.com/nar/article/49/21/e126/6382386#304709968

```
@article{zorrilla2021metagem,
  title={metaGEM: reconstruction of genome scale metabolic models directly from metagenomes},
  author={Zorrilla, Francisco and Buric, Filip and Patil, Kiran R and Zelezniak, Aleksej},
  journal={Nucleic acids research},
  volume={49},
  number={21},
  pages={e126--e126},
  year={2021},
  publisher={Oxford University Press}
}
```
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by franciscozorrilla** [comment]
@BiocondaBot please add label

**Comment by franciscozorrilla** [comment]
@BiocondaBot please fetch artifacts

File: recipes/metagem/post-link.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+pip install --user memote carveme smetana
```
Comment (bgruening) [inline recipes/metagem/post-link.sh:3]: you can move this one-liner directly into the meta.yml as `script`

Comment (franciscozorrilla) [reply recipes/metagem/post-link.sh]: thanks!

**Review by bgruening — APPROVED** [review]
Cool, thanks @franciscozorrilla

**Comment by franciscozorrilla** [comment]
@bgruening could you help with the following error message? Cant seem to fix this

```
13:44:03 BIOCONDA INFO (OUT) 0 location(s) to search for versions of memote:
13:44:03 BIOCONDA INFO (OUT) Given no hashes to check 0 links for project 'memote': discarding no candidates
13:44:03 BIOCONDA INFO (OUT) ERROR: Could not find a version that satisfies the requirement memote (from versions: none)
13:44:03 BIOCONDA INFO (OUT) ERROR: No matching distribution found for memote

```

File: recipes/metagem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  url: "https://github.com/franciscozorrilla/metaGEM/archive/refs/tags/v1.0.5.tar.gz"
+  sha256: "8ee8375967920fe9fabe276512f1dc2d663a8a85c73f28c534306fd66e4c3068"
+
+build:
+  noarch: generic
+  number: 0
+  script: "{{ PYTHON }} -m pip install --user -vvv memote carveme smetana"
```
Comment (bgruening) [inline recipes/metagem/meta.yaml:15]: Why not simply ` script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv`

Comment (franciscozorrilla) [reply recipes/metagem/meta.yaml:15]: I guess Im not sure how to specify that these 3 packages (memote carveme smetana) need to be installed via pip, while the ones below via conda?

Comment (bgruening) [reply recipes/metagem/meta.yaml:15]: Oh I see. No this is not recommended :)
We need to build those three packages at first as conda packages.

Use grayscull to build packages for those 3 dependencies: https://conda-forge.org/blog/posts/2020-03-05-grayskull/

(and if they are not bio-related submit them please to conda-froge https://github.com/conda-forge/staged-recipes/, ping me there, so I can help)

Comment (franciscozorrilla) [reply recipes/metagem/meta.yaml]: ok I see, thanks! I just checked and carveme appears to be on conda already, but will see what I can do about the other two. for now I will leave these tools out of the recipe and add them later

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@BiocondaBot please merge

**Pull request merged**